Automattic/wp-calypso
# Themes: Remove search as you type feature

When searching for a theme, the search will be triggered when **pressing Enter** instead of searching as you type. Furhter context can be found here pdgAPm-uo-p2#comment-446

#### Proposed Changes

* Change the existing `Search` component to accept a new prop `searchMode` that will derive the behaviour of the component. Now it could be either _search as you type_ or _search on enter_
* The default behaviour when the prop is not passed will be the previous one _`search as you type`_ so existing components that use this `Search` component should not be altered
* The component `ThemesMagicSearchCard` has changed to use this new _search on enter_ functionality

#### The following is a video with the new behaviour in the Theme search:


https://user-images.githubusercontent.com/3519124/193288972-8263daec-5fe2-4561-92b1-5ed9b4a3d448.mp4




#### Testing Instructions



* Navigate to `/themes/<site_id>`
* In the search input, insert a search term that returns results and press Enter, e.g. `Blank`
* Results should be filtered by the search term
* Search for a term that does not return results and press Enter, e.g. `Astra`
* A placeholder should be displayed with the message `Sorry, no themes found.`
* In the search input, click on the ✖️ icon on the right-hand-side of the input.
* The filter is removed and all results are displayed
* Add a search term and wait for the results to be filtered
* Press the `Escape` key
* The filter is removed and all results are displayed
* Add a search term and wait for the results to be filtered
* Delete the search term but do not press `Enter`
* The filter is removed and all results are displayed

#### Regression testing
- On the previous screen or in the home page of a selected site, click on the `Switch site` on the top-left corner
- ![CleanShot 2022-09-30 at 16 20 13@2x](https://user-images.githubusercontent.com/3519124/193290688-2eaf24b9-3f5d-4473-93fa-fcf340a5fa14.png)
- Enter a search term in the search input that appears and make sure the filter is applied `as you type`.


#### Pre-merge Checklist






Fixes #68281
